tabular/src/autogluon/tabular/models/tabular_nn/torch/tabular_nn_torch.py

```diff
@@ -562,9 +562,34 @@ def _get_default_resources(self):
         num_gpus = 0
         return num_cpus, num_gpus
 
+    def save(self, path: str = None, verbose=True) -> str:
+        import torch
+        # Save on CPU to ensure the model can be loaded on a box without GPU
+        self.model = self.model.to(torch.device("cpu"))
+        path = super().save(path, verbose)
+        # Put the model back to the device after the save
+        self.model.to(self.device)
+        return path
+
     @classmethod
     def load(cls, path: str, reset_paths=True, verbose=True):
+        import torch
+
         model: TabularNeuralNetTorchModel = super().load(path=path, reset_paths=reset_paths, verbose=verbose)
+
+        original_device_type = model.device.type
+        if 'cuda' in original_device_type:
+            device = torch.device(original_device_type if torch.cuda.is_available() else 'cpu')
+        elif 'mps' in original_device_type:
+            device = torch.device(original_device_type if torch.backends.mps.is_available() else 'cpu')
```

Comment:
This will force using GPU even if the original model was only trained on CPU. This should be documented

Reply:
no, it will use the original device if it is available; otherwise CPU